data/shared/citizen/scripting/v8/timer.js

```diff
@@ -72,30 +112,44 @@
         const localGameTime = Citizen.getTickCount(); // ms
         let i;
 
-        for (const id in timers) {
-            const timer = timers[id];
+        // Process timers
+        for (const timerId in timers) {
+            const timer = timers[timerId];
 
             if ((localGameTime - timer.lastRun) > timer.interval) {
                 try {
                     timer.callback();
-                } catch(e) {
-                    console.error('Unhandled error: ' + e.toString() + '\n' + e.stack);
+                } catch (e) {
+                    printError('timer', e);
                 }
 
                 timer.lastRun = localGameTime;
             }
         }
 
         // Process tickers
-        if (tickers.length > 0) {
-            i = tickers.length;
+        for (const tickerId in tickers) {
+            const ticker = tickers[tickerId];
 
-            while (i--) {
-                try {
-                    tickers[i]();
-                } catch(e) {
-                    console.error('Unhandled error: ' + e.toString() + '\n' + e.stack);
-                }
+            // If last call of ticker returned a promise,
+            // then we should wait for it
+            if (ticker.promise !== null) {
+                continue;
+            }
+
+            let result;
+            try {
+                result = ticker.callback();
+            } catch (e) {
+                printTickerError(e);
+                continue;
+            }
+
+            // We've got a promise!
+            if (result.then) {
```

Comment:
Using `typeof result.then === 'function'` is going to be way more reliable at determining whether the result is a Promise or not.

Reply:
It probably wouldn't be a bad idea to include this in the `try`/`catch` statement above, or it's own `try`/`catch` statement to avoid dying if you actually haven't got a promise.